pymatgen/symmetry/analyzer.py

```diff
@@ -1697,3 +1700,543 @@ def __str__(self):
+                         Residue: Set of species, fractional coordinates 
+                         and magnetic moments that are distinct. 
+                         Empty set if boolean is True.
+        """
+        same = False
+        nsite = len(struc1.sites)
+        if nsite != len(struc2.sites):
+            res = "number of sites differ"
+            pass
+        else:
+            sites1_prim = [np.mod(i.frac_coords + 10 + 1e-9, 1) - 1e-9 for i in struc1._sites]
+            sites1_prim = [tuple(k) for k in sites1_prim]
+            sites2_prim = [np.mod(i.frac_coords + 10 + 1e-9, 1) - 1e-9 for i in struc2._sites]
+            sites2_prim = [tuple(k) for k in sites2_prim]
+        
+            magms1_cart = [i.properties["magmom"].moment for i in struc1._sites]
+            magms1_cart = [tuple(k.round(3)) for k in magms1_cart]
+            magms2_cart = [i.properties["magmom"].moment for i in struc2._sites]
+            magms2_cart = [tuple(k.round(3)) for k in magms2_cart]
+            species1 = [i._species for i in struc1._sites]
+            species2 = [i._species for i in struc2._sites]
+      
+            set1 = set(tuple([tuple([species1[i],sites1_prim[i],magms1_cart[i]]) for i in range(nsite)]))
+            set2 = set(tuple([tuple([species2[i],sites2_prim[i],magms2_cart[i]]) for i in range(nsite)]))
+            res = self._difference(set1, set2, pos_tolerance=pos_tolerance, magmom_tolerance=magmom_tolerance)
+            if res == set():
+                same = True
+      
+        return same, res
+
+    def _difference(self, sitesmagms_set, sitesmagms_set_i, pos_tolerance=0.0015, magmom_tolerance=0.3):
+        """
+        Returns the difference between two sets of sites representing magnetic structures.
+
+        Takes:
+            sitesmagms_set1  set(tuple(tuple(str), tuple(float, float, float), tuple(float, float, float)))
+            sitesmagms_set2  set(tuple(tuple(str), tuple(float, float, float), tuple(float, float, float)))
+            pos_tolerance    float (default: 0.0015)
+            magmom_tolerance float (default: 0.3)
+
+        Returns:
+            set(tuple(tuple(str), tuple(float, float, float), tuple(float, float, float)))
+        """
+        res_sitesmagms_i = sitesmagms_set_i - sitesmagms_set
+        res_sitesmagms_new = sitesmagms_set - sitesmagms_set_i
+        res = []
+        for t1 in res_sitesmagms_new:
+            t1_equal2_t2 = False
+            for t2 in res_sitesmagms_i:
+                if t1[0] == t2[0]:
+                    if np.all(np.mod(np.abs(np.array(t1[1]) - np.array(t2[1])), 1) <= pos_tolerance):
+                        if np.all(np.abs(np.array(t1[2]) - np.array(t2[2])) <= magmom_tolerance):
+                            t1_equal2_t2 = True
+            if not t1_equal2_t2: 
+                res.append(t1)
+        return set(res) 
+
+    def _search4MagneticSpacegroup(self):
```

Comment:
Following PEP8, and to be more explicit, this could be called `_search_for_magnetic_spacegroup`